components/create-collective/NewCreateCollective.js

```diff
@@ -0,0 +1,223 @@
+    intl: PropTypes.object.isRequired,
+    createCollectiveV2: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      collective: {},
+      result: {},
+      category: null,
+      github: null,
+      form: false,
+      error: null,
+    };
+    this.createCollective = this.createCollective.bind(this);
+    this.messages = defineMessages({
+      'host.apply.title': {
+        id: 'host.apply.title',
+        defaultMessage: 'Apply to create a new {hostname} collective',
+      },
+      'collective.create.title': {
+        id: 'collective.create.title',
+        defaultMessage: 'Create an Open Collective',
+      },
+      'collective.create.description': {
+        id: 'collective.create.description',
+        defaultMessage: 'The place for your community to collect money and share your finance in full transparency.',
+      },
+    });
+
+    this.host = props.host;
+  }
+
+  componentDidMount() {
+    const { query } = this.props;
+    if (query.category === 'opensource' || query.token) {
+      this.setState({ category: 'opensource' });
+      if (query.step === 'form') {
+        this.setState({ form: true });
+      }
+      if (!query.step) {
+        this.setState({ form: false });
+      }
+    } else if (query.category === 'community') {
+      this.setState({ category: 'community' });
+    } else if (query.category === 'climate') {
+      this.setState({ category: 'climate' });
+    } else if (!query.category) {
+      this.setState({ category: null });
+    }
+    return;
+  }
+
+  componentDidUpdate(oldProps) {
+    const { query } = this.props;
+    if (oldProps.query.step !== query.step) {
+      if (query.step === 'form') {
+        this.setState({ form: true });
+      } else {
+        this.setState({ form: false });
+      }
+    }
+    if (oldProps.query.category !== query.category) {
+      if (query.category === 'opensource' || query.token) {
+        this.setState({ category: 'opensource' });
+      } else if (query.category === 'community') {
+        this.setState({ category: 'community' });
+      } else if (query.category === 'climate') {
+        this.setState({ category: 'climate' });
+      } else if (!query.category) {
+        this.setState({ category: null });
+      }
+    }
+    return;
+  }
+
+  handleChange(key, value) {
+    this.setState({
+      [key]: value,
+    });
+  }
+
+  async createCollective(CreateCollectiveInputType) {
+    // check we have agreed to TOS
+    if (!CreateCollectiveInputType.tos) {
+      this.setState({
+        error: 'Please accept the terms of service',
+      });
+      return;
+    }
+
+    // set state to loading
+    this.setState({ status: 'loading' });
+
+    // prepare object
+    CreateCollectiveInputType.tags = this.state.category;
+    if (this.state.github) {
+      CreateCollectiveInputType.githubHandle = this.state.github.handle;
+      this.host = { slug: 'opensource' };
```

Comment:
As discussed we will not have to set the host. in this case in the future, API changes coming shortly.

Reply:
Thanks!